src/Microsoft.AspNetCore.JsonPatch/Helpers/ICaseInsensitiveDictionary.cs

```diff
@@ -0,0 +1,14 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.AspNetCore.JsonPatch.Helpers
+{
+    internal interface ICaseInsensitiveDictionary
```

Comment:
This smells fishy - let's talk through why this is needed and what better solutions we might find